src/FSharp.Control.TaskSeq/TaskSeqInternal.fs

```diff
@@ -908,13 +891,11 @@ module internal TaskSeqInternal =
                 go <- step
         }
 
-        interface IAsyncDisposable with
-            override _.DisposeAsync() =
-                if not (isNull _rwLock) then
+        interface IDisposable with
```

Comment:
this shuts up the code suggestion about removing the `new`, and makes me more confident that it'll actually Dispose
@abelbraaksma So, what kind of Chesterton's fence have I fallen upon?

Reply:
This is a bug in F# (i.e., not raising the warning), we should not fix the mistake in the library by implementing an extra interface, we should fix F#@. Some very old versions of this code used to have it.

Reply:
But I'll double-check the "best practices" here. I think it's bad practice to offer synchronous disposal of async resources, but let's check.

This fix can lead to backward compat issues, so we need to take it out for sure.

Reply:
See #227, that's where this went.